TLM/TLM/Patch/_Vehicle/UnspawnPatch.cs

```diff
@@ -1,24 +1,13 @@
-using ColossalFramework;
-using ColossalFramework.Math;
-using CSUtil.Commons;
 using HarmonyLib;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using UnityEngine;
-
+using TrafficManager.Manager.Impl;
 namespace TrafficManager.Patch._Vehicle {
-    // TODO [issue #864] replace custom AI with harmony patch when possible.
-
-    //[HarmonyPatch(typeof(Vehicle), "Unspawn")]
-    //public static class UnspawnPatch {
-    //	/// <summary>
-    //	/// Notifies the vehicle state manager about a despawned vehicle.
-    //	/// </summary>
-    //	//[HarmonyPrefix]
-    //	public static void Prefix(ushort vehicleID) {
-    //		Constants.ManagerFactory.VehicleStateManager.OnDespawnVehicle(vehicleID, ref Singleton<VehicleManager>.instance.m_vehicles.m_buffer[vehicleID]);
-    //	}
-    //}
+    [HarmonyPatch(typeof(Vehicle), nameof(Vehicle.Unspawn))]
+    public static class UnspawnPatch {
+        /// <summary>
+        /// Notifies the vehicle state manager about a spawned vehicle.
```

Comment:
s/spawned/despawned

Reply:
I don't understand why this is not in the feature branch already. This was pushed in from a previous PR.
Nice catch though :)